ipaperftest.py

```diff
@@ -0,0 +1,485 @@
+#!/usr/bin/env python3
+
+import subprocess as sp
+import click
+import time
+import os
+import math
+
+VAGRANTFILE_TEMPLATE = """
+    Vagrant.configure("2") do |config|
+        {machine_configs}
+    end
+"""
+
+MACHINE_CONFIG_TEMPLATE = """
+    config.vm.define :{machine_name} do |{machine_name}|
+        {machine_name}.vm.provider "libvirt" do |v|
+            v.memory = {memory_size}
+            v.cpus = {cpus_number}
+        end
+        {machine_name}.vm.box = "{box}"
+        {machine_name}.vm.hostname = "{hostname}"
+        {machine_name}.vm.network "public_network", dev: "{net_interface}"
+        {machine_name}.vm.synced_folder ".", "/vagrant", disabled: true
+        {machine_name}.vm.provision "shell", inline: <<-SHELL
+            {extra_commands}
+        SHELL
+    end
+"""
+
+HOSTS_FILE_TEMPLATE = """
+[ipaserver]
+server
+
+[ipaserver:vars]
+ipaadmin_password=password
+ipadm_password=password
+ipaserver_domain={domain}
+ipaserver_realm={realm}
+ipaserver_setup_dns=yes
+ipaserver_no_forwarders=yes
+
+[ipaclients]
+{client_hostnames}
+
+[ipaclients:vars]
+ipaadmin_password=password
+ipaserver_domain={domain}
+ipaserver_realm={realm}
+"""
+
+ANSIBLE_CFG_TEMPLATE = """
+[defaults]
+deprecation_warnings=False
+roles_path   = {cwd}/ansible-freeipa/roles
+library      = {cwd}/ansible-freeipa/plugins/modules
+module_utils = {cwd}/ansible-freeipa/plugins/module_utils
+"""
+
+ANSIBLE_FETCH_FILES_PLAYBOOK = """
+---
+- name: Fetch IPA server log files
+  hosts: ipaserver
+  become: true
+  tasks:
+    - synchronize:
+        src: "{{{{ item }}}}"
+        dest: "{cwd}/sync/{{{{ ansible_hostname }}}}/"
+        mode: pull
+        use_ssh_args: yes
+      with_items:
+        - "/var/log/ipaserver-install.log"
+        - "/var/log/ipaclient-install.log"
+        - "/var/log/httpd"
+        - "/var/log/dirsrv"
+        - "/var/log/krb5kdc.log"
+        - "/var/log/pki/pki-tomcat/ca/transactions"
+        - "/home/vagrant/saroutput"
+
+- name: Fetch IPA clients log files
+  hosts: ipaclients
+  become: true
+  tasks:
+    - synchronize:
+        src: "{{{{ item }}}}"
+        dest: "{cwd}/sync/{{{{ ansible_hostname }}}}/"
+        mode: pull
+        use_ssh_args: yes
+      with_items:
+        - "/var/log/ipaclient-install.log"
+
+"""
+
+
+class EnrollmentTest:
```

Comment:
Including the tests in one file is fine for starters IMHO. We can break them out into discoverable plugins later.